source/frontends/common2/programoptions.h

```diff
@@ -71,7 +71,7 @@ namespace common2
     std::vector<std::string> natPortFwds;
   };
 
-  enum class OptionsType { applen, sa2 };
+  enum class OptionsType { applen, sa2, mariani };
```

Comment:
Not worse than it already is, but I don't fancy a big inheritance for this.

Reply:
Yeah, this one is just aesthetics for me. macOS apps don't really use command-line arguments, the new enum would just let me explicitly say that it's not actually supporting sa2's arguments.

Alternatively, a `none` value might be better. Your other frontends can then all use that to say they don't have frontend-specific arguments.

Reply:
I suspect the latter.
This should work in linux too.

https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/fcntl.2.html#//apple_ref/doc/man/2/fcntl

I will test it and propose a PR.

Reply:
> Alternatively, a `none` value might be better. Your other frontends can then all use that to say they don't have frontend-specific arguments.

I prefer `none` then if you don't need command line arguments.

Reply:
`none` is fine with me.

But alternatively, perhaps `EmulatorOptions::getEmulatorOptions()` can call an optional virtual function that sa2 and applen can implement a subclass and handle arguments specific to them?

Reply:
Put `none` for now, then I will try to design the subclass without going crazy.